src/thirdparty/caman/instagram.js

```diff
@@ -0,0 +1,134 @@
+
+
+$(function() {
+  "use strict";
+
+  var CommandManager = brackets.getModule("command/CommandManager");
+  var Commands = brackets.getModule("command/Commands");
+  var fs = brackets.getModule("fileSystemImpl");
+  var Buffer = brackets.getModule("filesystem/impls/filer/BracketsFiler").Buffer;
+
+  function savePhoto(){
+    Caman(canvas, function () {
+      this.render(function () {
+            var image = this.toBase64();
+            var binaryDataStr = /^data:image\/png;base64,(.+)/.exec(image)[1];
+            //self.camera.savePhoto(base64ToBuffer(binaryDataStr));
+              var binary = window.atob(binaryDataStr);
+              var len = binary.length;
+              var bytes = new Uint8Array(len);
+              for(var i = 0; i < len; i++) {
+                bytes[i] = binary.charCodeAt(i);
+              }
+              var data = new Buffer(bytes.buffer);
+              fs.writeFile(savePath, data, {encoding: null}, function() {
+                // Update the file tree to show the new file
+                CommandManager.execute(Commands.FILE_REFRESH);
+                console.log("save path: " + savePath);
+              });
+           console.log("attributes: " + canvas.width + ", " + canvas.height);
+         });
+       });
+       console.log("photo saved");
+  }
+
+  /*
+  Can't use this pattern at the moment since the Caman function won't run before
```

Comment:
It means you need to add an event for the DOM to be loaded, maybe use `DOMContentLoaded`